### PR TITLE
Add igraph [redux]

### PR DESCRIPTION
**Pull request by sodre**
This is the cleaned up version of PR #1385.

### CODE REVIEW

File: recipes/igraph/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: http://igraph.org/nightly/get/c/{{ name }}-{{ version }}.tar.gz
+  {{ hash_type }}: {{ hash_val }} 
+  patches:
+    - b215195214abb8355121a8430c86df04e1f00eae.patch  # fixes igraph/igraph#906
+
+build:
+  skip: True  # [win or osx]
```
Comment (jakirkham) [inline recipes/igraph/meta.yaml]: Why no OS X?

File: recipes/igraph/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - pkg-config
+  run:
+    - gmp 
+    - libxml2
+
+test:
+  files:
+    - igraph_test.c
+  requires:
+    - pkg-config
```
Comment (jakirkham) [inline recipes/igraph/meta.yaml]: Could we add some file tests too (e.g for headers)?

File: recipes/igraph/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+about:
+  home: http://igraph.org
+  license: GPL2
```
Comment (jakirkham) [inline recipes/igraph/meta.yaml]: Optionally can add `license_family: GPL`.

File: recipes/igraph/meta.yaml (hunk 1 above)
Comment (sodre) [inline recipes/igraph/meta.yaml]: There is a compilation issue in OS X. My hope is to get the linux version merged first and add the win and OS X versions afterwards.

File: recipes/igraph/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  license_file: COPYING
+  summary: "igraph is a collection of network analysis tools with the emphasis on efficiency, portabiliy and ease of use. igraph is open source and free. igraph can be programmed in R, Python and C/C++."
```
Comment (jakirkham) [inline recipes/igraph/meta.yaml]: Could we shorten this line? Info of this length can go in `description`, but should be [wrapped]( https://github.com/conda-forge/staged-recipes/blob/99c66d02f87f1bcc5eaaa7b835b833430a9cfc19/recipes/example/meta.yaml#L58-L63 ).

File: recipes/igraph/meta.yaml (hunk 2 above)
Comment (sodre) [inline recipes/igraph/meta.yaml]: igraph_test.c is being compiled by run_test.sh. That is also part of the actual igraph unit tests. Shouldn't that be sufficient?

**Comment by scopatz** [comment]
Thanks so much @sodre! Sorry about the excessive delay. Just keep on us in the future.

**Pull request merged**